src/Core/System/StateMachine/StateMachineRegistry.php

```
@@ -252,12 +257,12 @@ private function getTransitionDestinationById(string $stateMachineName, string $
     {
         $stateMachine = $this->getStateMachine($stateMachineName, $context);
 
-        /** @var StateMachineTransitionCollection $stateMachineTransitions */
         $stateMachineTransitions = $stateMachine->getTransitions();
+        \assert($stateMachineTransitions !== null);
```

should we throw an exception instead?

---

I would be reluctant to change anything about behavior with this PR.